compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -923,7 +923,51 @@ trait Implicits:
           // example where searching for a nested type causes an infinite loop.
           None
 
-    MissingImplicitArgument(arg, pt, where, paramSymWithMethodCallTree, ignoredInstanceNormalImport)
+    def allImplicits(currImplicits: ContextualImplicits): List[ImplicitRef] =
+      if currImplicits.outerImplicits == null then currImplicits.refs
+      else currImplicits.refs ::: allImplicits(currImplicits.outerImplicits)
+
+    /** Ensure an implicit is not a Scala 2-style implicit conversion, based on its type */
+    def isScala2Conv(typ: Type): Boolean = typ match {
+      case PolyType(_, resType) => isScala2Conv(resType)
+      case mt: MethodType => !mt.isImplicitMethod && !mt.isContextualMethod
+      case _ => false
+    }
+
+    def hasErrors(tree: Tree): Boolean =
+      if tree.tpe.isInstanceOf[ErrorType] then true
+      else
+        tree match {
+          case Apply(_, List(arg)) => arg.tpe.isInstanceOf[ErrorType]
+          case _ => false
+        }
+
+    def ignoredConvertibleImplicits = arg.tpe match
+      case fail: SearchFailureType =>
+        if (fail.expectedType eq pt) || isFullyDefined(fail.expectedType, ForceDegree.none) then
+          // Get every implicit in scope and try to convert each
+          allImplicits(ctx.implicits)
+            .distinctBy(_.underlyingRef.denot)
+            .view
+            .map(_.underlyingRef)
+            .filter { imp =>
+              if isScala2Conv(imp.underlying) || imp.symbol == defn.Predef_conforms then
+                false
+              else
+                // Using Mode.Printing will stop it from printing errors
+                val tried = Contexts.withMode(Mode.Printing) {
+                    typed(
+                      tpd.ref(imp).withSpan(arg.span),
+                      fail.expectedType,
+                      ctx.typerState.ownedVars
+                    )
+                  }
+                !hasErrors(tried) && fail.expectedType =:= tried.tpe
```

Comment:
Extracting this to a separate method would make this piece of code much more readable